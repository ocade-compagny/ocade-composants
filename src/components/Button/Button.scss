@import "../config.scss";

.btn {
    border-radius: $border-radius;
    background-color: orange;
}