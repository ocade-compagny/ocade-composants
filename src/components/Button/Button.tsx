import React from 'react'

export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode
  variant: `contained` | `outlined` | `text`
  shape?: boolean
}

export const Button: React.FC<ButtonProps> = ({
  children,
  variant,
  shape,
  ...props
}) => {
  const classNames = `
    btn 
    ${ variant ? `btn-${variant}` : `` } 
    ${ shape ? "btn-rounded" : `` }
  `;
  return (
      <button className={classNames} {...props}>
        {children}
      </button>
  )
}
