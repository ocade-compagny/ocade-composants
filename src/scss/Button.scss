/* base styling for our button */
.btn {
  font-family: $font1;
  color: black;
  background-color: white;

  &.btn-rounded { border-radius: $shape; }

  &.btn-contained { border: 1px solid rgb(207 60 60); }
  &.btn-outlined { border: 1px solid rgb(60 207 104); }
  &.btn-text { border: 1px solid rgb(60 62 207); }
}
