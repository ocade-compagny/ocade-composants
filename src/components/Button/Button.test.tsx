import renderer from 'react-test-renderer'
import { expect } from '@jest/globals'
import { Button } from './Button'

it('changes the class when hovered', () => {
  const component = renderer.create(
    <Button variant="contained">click Me</Button>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
