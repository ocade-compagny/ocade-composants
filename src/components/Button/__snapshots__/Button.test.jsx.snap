// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<button
  className="btn btn-undefined btn-undefined"
>
  click Me
</button>
`;

exports[`changes the class when hovered 2`] = `
<button
  className="btn btn-undefined btn-undefined"
>
  click Me
</button>
`;

exports[`changes the class when hovered 3`] = `
<button
  className="btn btn-undefined btn-undefined"
>
  click Me
</button>
`;
